src/autoscaler/metricsgateway/nozzle.go

```diff
@@ -121,7 +121,7 @@ func (n *Nozzle) filterEnvelopes(envelops []*loggregator_v2.Envelope) {
 					n.envelopChan <- e
 				}
 			case *loggregator_v2.Envelope_Timer:
-				if e.GetTimer().GetName() == "http" {
+				if e.GetTimer().GetName() == "http" && e.GetDeprecatedTags()["peer_type"].GetText() == "Client" {
```

Comment:
@silvestre 
**1** According to the document [peertype](https://github.com/cloudfoundry/dropsonde-protocol/blob/ce0e609a7a57f041b74e661f13d32e5f7a673a5d/events/README.md#peertype).

Name | Number | Description
-- | -- | --
Client | 1 | Request is made by this process.
Server | 2 | Request is received by this process.

`e.GetDeprecatedTags()["peer_type"].GetText() == "Client" ` means we only care the requests made by the application. I think we should count the requests received by the application, right?
**2** And  method GetDeprecatedTags() contains "Deprecated", so maybe one day there is no "peer_type" tag so the nozzle will skip all http event?

Reply:
@qibobo : Client/server is from the viewpoint of the application emitting the event, in this case the `gorouter`. We should only count requests made by the gorouter to the application, when it has decided which app instance to target. The other event is emitted when the request is received and has no information on which app instance will process the request.

With regards to the deprecation, I had assumed you had already discussed this with the routing team, as just two lines above we already check for the origin which is also in the deprecated tags: https://github.com/cloudfoundry/app-autoscaler/blob/a911a395ac4a2caede4b63adccc8eb6dc0b32fa5/src/autoscaler/metricsgateway/nozzle.go#L119

I will ping the routing team to get their thoughts on the deprecation strategy.

Reply:
Raised the deprecation question in https://cloudfoundry.slack.com/archives/CFX13JK7B/p1574065292072500

Reply:
@silvestre 
How about to check wether the "peer_type" tag exists first and then check the value? So that when it is removed, it will not break the feature.

Reply:
@qibobo Good idea, implemented in 4f5e3aa24e7db8af1aae9bb96e73c6642b7ba70b